app/views/contexts/calculated_summary_context.py

```diff
@@ -30,78 +35,101 @@ def build_groups_for_section(self, section, return_to_block_id):
+        block: ImmutableDict = self._schema.get_block(block_id)  # type: ignore
 
         calculated_section = self._build_calculated_summary_section(
             block, current_location
         )
         calculation = block["calculation"]
 
-        groups = self.build_groups_for_section(calculated_section, return_to_block_id)
+        groups = self.build_groups_for_section(
+            calculated_section, block_id, current_location
+        )
 
         formatted_total = self._get_formatted_total(
             groups or [],
             current_location=current_location,
-            calculation_operator=ValueSourceResolver.get_calculation_operator(
+            calculation=ValueSourceResolver.get_calculation_operator(
                 calculation["calculation_type"]
-            ),
+            )
+            if calculation.get("answers_to_calculate")
+            else calculation["operation"],
         )
 
-        context = {
+        collapsible = block.get("collapsible") or False
+        block_title = block.get("title")
+
+        return {
             "summary": {
                 "groups": groups,
                 "answers_are_editable": True,
                 "calculated_question": self._get_calculated_question(
                     calculation, formatted_total
                 ),
-                "title": block.get("title") % dict(total=formatted_total),
-                "collapsible": block.get("collapsible", False),
+                "title": block_title % dict(total=formatted_total)
+                if block_title
+                else None,
+                "collapsible": collapsible,
                 "summary_type": "CalculatedSummary",
             }
         }
 
-        return context
-
-    def _build_calculated_summary_section(self, rendered_block, current_location):
+    def _build_calculated_summary_section(
+        self, rendered_block: ImmutableDict, current_location: Location
+    ) -> Mapping:
         """Build up the list of blocks only including blocks / questions / answers which are relevant to the summary"""
-        section_id = self._schema.get_section_id_for_block_id(current_location.block_id)
-        group = self._schema.get_group_for_block_id(current_location.block_id)
-        blocks = []
-        answers_to_calculate = rendered_block["calculation"]["answers_to_calculate"]
-        blocks_to_calculate = [
-            self._schema.get_block_for_answer_id(answer_id)
-            for answer_id in answers_to_calculate
-        ]
-        unique_blocks = list(
-            {block["id"]: block for block in blocks_to_calculate}.values()
-        )
-
-        for block in unique_blocks:
-            if QuestionnaireSchema.is_question_block_type(block["type"]):
-                transformed_block = self._remove_unwanted_questions_answers(
-                    block, answers_to_calculate, current_location=current_location
+        if (block_id := current_location.block_id) and (
```

Comment:
Same here, this indentation seems to be for the sake of type hints. Should be removed.

Reply:
Updated 👍